### DISCUSSION TITLE
LED Matrix schaltet sich eigenständig wieder ein

### DISCUSSION THREAD

**Discussion by msw-home** [pybricks/support#1795] [Q&A]
Hallo, merkwürdiges Verhalten der LED Matrix.
Ich stecke das Hub Ladekabel ein und schalte den Hub aus.
LED Matrix schaltet sich korrekt aus,
aber 1 Sekunde später wieder ein, mit grüner Farbe!?
Passiert nicht, wenn das Ladekabel nicht angeschlossen ist. 

https://github.com/user-attachments/assets/43dcb100-351d-4e8d-9960-160a06d9f59b

**Comment by BertLindeman** [comment]
Manfred, kein Ahnung.

Maybe it helps to show the firmware level of the hub?
A python program:
```python

from pybricks import version

print(f'version {version}')
```

No 3x3 light matrix here. (A bit expensive just to test with it)

Bert

**Reply by BertLindeman** [reply]
Manfred,

Another idea: does the behavior change if you use another port for the matrix?

Is the program a simple program you can share here?

I tried something similar with a LED-light, but that is obviously another case.

Bert

**Reply by dlech** [reply]
The problem is that I/O ports aren't powered down when the hub is "shut down". For hubs with USB chargers, the CPU has to keep running to control the charger and the charge indication light, so we can't actually power down the hub while charging. But we should be able to power down the I/O ports completely.

It looks like we are shutting down the voltage on pin 4 (i.e. a BOOST color/distance sensor won't light up), but not shutting down the power on pins 1 and 2 for non-motor devices that need extra power.

**Reply by msw-home** [reply]
👍👍👍

**Comment by msw-home** [comment]
Hallo Bert,

 

hat nichts mit dem Port zu tun, immer gleich.

Ist aber kein Pybricks Problem!

Verhält sich auch so, wenn ich die Original Firmware von Lego wieder drauf spiele.

 

Mit freundlichen Grüßen

 

 

Manfred Schmidt-Winterstein

Wacholderweg 72A

38239 Salzgitter

 

Tel:    0151 26091955

Mail:   ***@***.***> ***@***.***

 

Von: Bert ***@***.***> 
Gesendet: Saturday, 31 August 2024 16:59
An: pybricks/support ***@***.***>
Cc: msw-home ***@***.***>; Author ***@***.***>
Betreff: Re: [pybricks/support] LED Matrix schaltet sich eigenständig wieder ein (Discussion #1795)

 

Manfred,

Another idea: does the behavior change if you use another port for the matrix?

Is the program a simple program you can share here?

I tried something similar with a LED-light, but that is obviously another case.

Bert
